core/src/com/mygdx/assets/Assets.java

```diff
@@ -0,0 +1,81 @@
+package com.mygdx.assets;
+
+import com.badlogic.gdx.assets.AssetManager;
+import com.badlogic.gdx.assets.loaders.resolvers.InternalFileHandleResolver;
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.g3d.Model;
+import com.badlogic.gdx.utils.Array;
+import com.badlogic.gdx.utils.Disposable;
+import com.mygdx.assets.assetloaders.ArrayLoader;
+import com.mygdx.assets.assetloaders.ArrayLoader.ArrayLoaderParameter;
+import com.mygdx.assets.assetloaders.JsonLoader;
+import com.mygdx.config.EntityConfig;
+import com.mygdx.config.FieldConfig;
+import com.mygdx.config.GameConfigsService;
+import com.mygdx.game.util.Const;
+import lombok.NonNull;
+
+public class Assets implements Disposable {
+
+    @NonNull
+    private final AssetManager assetManager;
+
+    public Assets() {
+        this.assetManager = new AssetManager();
+        initCustomLoaders();
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private void initCustomLoaders() {
+        assetManager.setLoader(Array.class, new ArrayLoader(new InternalFileHandleResolver()));
+        assetManager.setLoader(FieldConfig.class, new JsonLoader<>(new InternalFileHandleResolver()));
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public void loadConfig() {
+        assetManager.load(Const.ASSETS_PATH + "entity_configs/field",
+                Array.class,
+                new ArrayLoaderParameter(FieldConfig.class, "json"));
+        assetManager.finishLoading();
+    }
+
+    public void loadAssets() {
+        loadModels(FieldConfig.class);
+        loadTextures();
+        assetManager.finishLoading();
+    }
+
+    public Model getModel(@NonNull EntityConfig config) {
+        var filename = assetManager.getAssetFileName(config);
+        var directory = filename.substring(0, filename.lastIndexOf("/") + 1);
+        return assetManager.get(directory + config.getName() + ".g3db");
```

Comment:
Tutaj trochę bardzo naokoło się dostaję do tego zasobu - konstruuję tutaj taką ścieżkę do tego pliku tak, żeby był w tym samym folderze co config. Inaczej mogłoby to być, gdyby te pliki znajdowały się wszystkie w tym samym folderze, bo nie robiłbym wtedy getAssetFilename tylko doklejał do czegoś z Const. + w jsonie jednak ta nazwa powinna już mieć w sobie końcówkę .g3db.